dev/deploy-to-container/cli.js

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env node
+
+import Docker from 'dockerode'
+import path from 'path'
+import fs from 'fs-extra'
+import * as tar from 'tar'
+import yargs from 'yargs/yargs'
+import { hideBin } from 'yargs/helpers'
+import slugify from 'slugify'
+import { nanoid, customAlphabet } from 'nanoid'
+import { alphanumeric } from 'nanoid-dictionary'
+
+const nanoidAlphaNum = customAlphabet(alphanumeric, 16)
+
+async function main () {
+  const basePath = process.cwd()
+  const releasePath = path.join(basePath, 'release')
+  const argv = yargs(hideBin(process.argv)).argv
+
+  // Parse branch argument
+  let branch = argv.branch
+  if (!branch) {
+    throw new Error('Missing --branch argument!')
+  }
+  if (branch.indexOf('/') >= 0) {
+    branch = branch.split('/').slice(1).join('-')
+  }
+  branch = slugify(branch, { lower: true, strict: true })
+  if (branch.length < 1) {
+    throw new Error('Branch name is empty!')
+  }
+  console.info(`Will use branch name "${branch}"`)
+
+  // Parse domain argument
+  const domain = argv.domain
+  if (!domain) {
+    throw new Error('Missing --domain argument!')
+  }
+  const hostname = `ma-${branch}.${domain}`
+  console.info(`Will use hostname "${hostname}"`)
+
+  // Connect to Docker Engine API
+  console.info('Connecting to Docker Engine API...')
+  const dock = new Docker()
+  await dock.ping()
+  console.info('Connected to Docker Engine API.')
+
+  // Extract release artifact
+  console.info('Extracting release artifact...')
+  if (!(await fs.pathExists(path.join(basePath, 'release.tar.gz')))) {
+    throw new Error('Missing release.tar.gz file!')
+  }
+  await fs.emptyDir(releasePath)
+  await tar.x({
+    cwd: releasePath,
+    file: 'release.tar.gz'
+  })
+  console.info('Extracted release artifact successfully.')
+
+  // Update the settings_local.py file
+  console.info('Setting configuration files...')
+  const mqKey = nanoidAlphaNum()
+  const settingsPath = path.join(releasePath, 'backend/mlarchive/settings/settings_sandbox.py')
+  const cfgRaw = await fs.readFile(path.join(basePath, 'dev/deploy-to-container/settings_sandbox.py'), 'utf8')
+  await fs.outputFile(settingsPath,
+    cfgRaw
+      .replace('__DBHOST__', `ma-db-${branch}`)
+      .replace('__SECRETKEY__', nanoid(36))
+      .replace('__MQCONNSTR__', `amqp://mailarchive:${mqKey}@ma-mq-${branch}/da`)
+      .replace('__HOSTNAME__', hostname)
+  )
+  await fs.copy(path.join(basePath, 'docker/scripts/app-create-dirs.sh'), path.join(releasePath, 'app-create-dirs.sh'))
+  await fs.copy(path.join(basePath, 'dev/deploy-to-container/start.sh'), path.join(releasePath, 'start.sh'))
+  await fs.copy(path.join(basePath, 'test/data'), path.join(releasePath, 'test/data'))
+  console.info('Updated configuration files.')
+
+  // Pull latest DB image
+  console.info('Pulling DB docker image...')
+  const dbImagePullStream = await dock.pull('postgres:14.6')
```

Comment:
We are using pg 16 in staging + prod. Should be changed to `postgres:16`?

Reply:
changed to 16